samples/excel/85-preview-apis/data-types-references.yaml

```diff
@@ -0,0 +1,337 @@
+script:
+  content: >
+    $("#setup").click(() => tryCatch(setup));
+    $("#add-entities-to-table").click(() => tryCatch(addEntitiesToTable));
+
+    async function addEntitiesToTable() {
+      await Excel.run(async (context) => {
+        // This method creates entity values with references to other entity values. 
+        // It retrieves data and references for each of the employees in the table
+        // in the worksheet, creates entity values for each of those employees, and
+        // adds the entities to the table.
+
+        const employeesTable = context.workbook.tables.getItem("EmployeesTable");
+
+        // Add a new column to the table for the entity values.
+        employeesTable.columns.getItemOrNullObject("Employee").delete();
+        const employeeColumn = employeesTable.columns.add(0, null, "Employee");
+
+        // Get employee data from the table.
+        const dataRange = employeesTable.getDataBodyRange();
+        dataRange.load("values");
+        await context.sync();
+
+        // Create the entities by mapping the employee IDs in table to the sample
+        // JSON employee data.
+        const entities = dataRange.values.map((rowValues) => {
+          // Get employee properties.
+          const employee = getEmployee(rowValues[1] /* Employee ID */);
+
+          // Collect all the references that are needed to make the employee entity.
+          // A reference is a link to another entity value outside the original entity.
+          const references = collectReferences(employee);
+
+          // Create employee entity from employee properties.
+          const employeeEntity = makeEmployeeEntity(
+            rowValues[2] /* Last name */,
+            rowValues[3] /* First name */,
+            employee,
+            references
+          );
+
+          // Collect reference values for the employee entity.
+          const referencedValues: any[] = [{ type: "Root" }];
+          references.slice(1).forEach((id) => {
+            const referencedEmployee = getEmployee(id);
+            referencedValues.push(
+              makeEmployeeEntity(referencedEmployee.lastName, referencedEmployee.firstName, referencedEmployee, references)
+            );
+          });
+          employeeEntity.referencedValues = referencedValues;
+
+          return [employeeEntity];
+        });
+
+        // Add the complete entities to the Employees Table.
+        employeeColumn.getDataBodyRange().valuesAsJson = entities;
+        employeeColumn.getRange().format.autofitColumns();
+        await context.sync();
+      });
+    }
+
+    // Collect all the references to include in the employee entity.
+    // A reference is a link to another entity value outside the original entity.
+    function collectReferences(employee) {
+      const references: number[] = [];
+      collectReferencesM(employee, references);
+      return references;
+    }
+
+    function collectReferencesM(employee, references: number[]) {
```

Comment:
@jakobpn What does the "M" stand for in the method name `collectReferencesM`?

Reply:
M stands for "Mark". So just to give it a different name from collectReferences.